pkg/fleet/installer/packages/pkg_manager.go

```diff
@@ -0,0 +1,55 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build !windows
+
+package packages
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os/exec"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+// removeDebRPMPackage removes a package installed via deb/rpm package manager
+// It doesn't remove dependencies or purge as we want to keep existing configuration files
+// and reinstall the package using the installer.
+func removeDebRPMPackage(ctx context.Context, pkg string) (err error) {
+	span, _ := tracer.StartSpanFromContext(ctx, "remove_deb_rpm_package")
+	defer func() { span.Finish(tracer.WithError(err)) }()
+	// Compute the right command depending on the package manager
+	var cmd *exec.Cmd
+	if _, pathErr := exec.LookPath("dpkg"); pathErr == nil {
+		cmd = exec.Command("dpkg", "-r", agentPackage)
+	} else if _, pathErr := exec.LookPath("rpm"); pathErr == nil {
+		// Check if package exist, else the command will fail
+		pkgErr := exec.Command("rpm", "-q", agentPackage).Run()
+		if pkgErr == nil {
+			cmd = exec.Command("rpm", "-e", "--nodeps", agentPackage)
```

Comment:
I wonder if we can hit some bad surprises in the prerm/postrm scripts of the different agent versions you can uninstall, is it possible to skip those scripts with the uninstall?

Reply:
ex [noscripts](http://ftp.rpm.org/max-rpm/s1-rpm-install-additional-options.html#S2-RPM-INSTALL-NOSCRIPTS)

Reply:
That's a good point; I agree and I'll add it.

For completeness I checked the prerm/postrm; there are a few caveats in not running them:
- Some files not managed by apt will be leftover in `/opt/datadog-agent`; such as the RC DB, custom integrations...
- The install method will be kept and the rest of the install will not override it - wrongly reporting dpkg / rpm

Reply:
The install method you're referring is the debug telemetry file?

Reply:
We use it in Fleet to check if an agent can be upgraded; so it's not just telemetry. Though it should be fine here as it's overwritten by the install script.